File: 102.binary-tree-level-order-traversal.js
// @leet start
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
	if (!root) return [];
	var queue = [root];
	var answer = [];
	while (queue.length > 0) {
		var length = queue.length;
		var list = [];
		for (let i = 0; i < length; i++) {
			node = queue.shift();
			if (node.left !== null) queue.push(node.left);
			if (node.right !== null) queue.push(node.right);
			list.push(node.val);
		}
		answer.push(list);
	}
	return answer;
};
// @leet end

File: 102.binary-tree-level-order-traversal.cpp
// @leet start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
#include <deque>
#include <vector>
class Solution {
public:
  vector<vector<int>> levelOrder(TreeNode *root) {
		if (!root) {
			return {};
		}
		std::deque<TreeNode*> q;
		q.push_back(root);
		std::vector<std::vector<int>> answer;
		while (q.size()) {
			int len = q.size();
			std::vector<int>row;
			for (int i = 0; i < len; i++) {
				TreeNode* node = q.front();
				q.pop_front();
				row.push_back(node->val);
				if(node->left)q.push_back(node->left);
				if(node->right)q.push_back(node->right);
			}
			answer.push_back(row);
		}
		return answer;
  }
};
// @leet end
